src/catalog/function_catalog.cpp

```diff
@@ -0,0 +1,355 @@
+          type::Type::BOOLEAN, bool_type_size, "secdef_column", true); //val9
+
+      auto leakproof_column = catalog::Column(
+          type::Type::BOOLEAN, bool_type_size, "leakproof_column", true); //val10
+
+      auto isstrict_column = catalog::Column(
+          type::Type::BOOLEAN, bool_type_size, "isstrict_column", true); //val11
+
+      auto retset_column = catalog::Column(
+          type::Type::BOOLEAN, bool_type_size, "retset_column", true); //val12
+    
+      auto volatile_column = catalog::Column(
+          type::Type::VARCHAR, varchar_type_size, "volatile_column", false); //val13
+
+      auto num_args_column = catalog::Column(
+          type::Type::INTEGER, integer_type_size, "num_params", true); //val14
+      num_args_column.AddConstraint(not_null_constraint);
+
+      //might need a not null constraint?
+      auto num_default_args_column = catalog::Column(
+          type::Type::INTEGER, integer_type_size, "num_default_params", true); //val15
+
+      auto rettype_oid_column = catalog::Column(
+          type::Type::INTEGER, integer_type_size, "rettype_oid", true); //val16
+      lang_oid_column.AddConstraint(not_null_constraint);
+
+      auto arg_types_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "arg_types", false);
+      name_column.AddConstraint(not_null_constraint); //val17
+
+
+      auto all_arg_types_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "all_arg_types", false); //val18
+
+      auto arg_modes_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "arg_modes", false); //val19
+
+      auto arg_names_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "arg_names", false); //val20
+
+
+      auto arg_defaults_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "arg_defaults", false); //21
+
+      auto src_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "src", false); //22
+      name_column.AddConstraint(not_null_constraint);
+
+      auto bin_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "bin", false); //23
+      
+      auto config_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "config", false); //24
+
+      auto aclitem_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "aclitem", false); //25
+
+      std::unique_ptr<catalog::Schema> function_catalog_schema(
+          new catalog::Schema({function_oid_column, name_column, owner_oid_column,
+            lang_oid_column, cost_column, rows_column, variadic_oid_column,isagg_column, iswindow_column, secdef_column, leakproof_column, isstrict_column,retset_column, volatile_column, num_args_column, num_default_args_column,rettype_oid_column, arg_types_column, all_arg_types_column, arg_modes_column, arg_names_column, arg_defaults_column, src_column, bin_column, config_column, aclitem_column}));
```

Comment:
Unfortunately the schema api requires you to input all the column names, and the pg_proc catalog has 20+ columns.